provider/resources.go

```diff
@@ -400,6 +400,9 @@ func preConfigureCallbackWithLogger(credentialsValidationRun *atomic.Bool, gcpCl
 			"CLOUDSDK_CORE_PROJECT",
 		})
 		if project == "" {
+			if project == "No Project" {
```

Comment:
This code will never run, since `project == ""` requires that `project != "No Project"`.

Reply:
Thank you, great catch.